library/rendering/entity_rendering/src/main/java/org/quiltmc/qsl/rendering/entity/api/client/ArmorRenderingRegistry.java

```diff
@@ -0,0 +1,94 @@
+import net.minecraft.entity.EquipmentSlot;
+import net.minecraft.entity.LivingEntity;
+import net.minecraft.item.ItemConvertible;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.Identifier;
+
+import org.quiltmc.qsl.base.api.event.Event;
+import org.quiltmc.qsl.rendering.entity.impl.client.ArmorRenderingRegistryImpl;
+
+@Environment(EnvType.CLIENT)
+public final class ArmorRenderingRegistry {
+	private ArmorRenderingRegistry() {
+		throw new UnsupportedOperationException("ArmorRenderingRegistry only contains static declarations.");
+	}
+
+	public static final Identifier DEFAULT_PHASE = Event.DEFAULT_PHASE;
+
+	public static void registerTextureProvider(@NotNull Identifier phaseIdentifier,
+			@NotNull TextureProvider provider, @NotNull ItemConvertible... items) {
+		for (var itemC : items) {
+			ArmorRenderingRegistryImpl.registerTextureProvider(itemC.asItem(), phaseIdentifier, provider);
+		}
+	}
+
+	public static void addTextureProviderPhaseOrdering(@NotNull Identifier firstPhase, @NotNull Identifier secondPhase,
+			@NotNull ItemConvertible... items) {
+		for (var itemC : items) {
+			ArmorRenderingRegistryImpl.addTextureProviderPhaseOrdering(itemC.asItem(), firstPhase, secondPhase);
+		}
+	}
+
+	public static void registerTextureProvider(@NotNull TextureProvider provider, @NotNull ItemConvertible... items) {
+		registerTextureProvider(DEFAULT_PHASE, provider, items);
+	}
+
+	public static void registerModelProvider(@NotNull Identifier phaseIdentifier,
+			@NotNull ModelProvider provider, @NotNull ItemConvertible... items) {
+		for (var itemC : items) {
+			ArmorRenderingRegistryImpl.registerModelProvider(itemC.asItem(), phaseIdentifier, provider);
+		}
+	}
+
+	public static void addModelProviderPhaseOrdering(@NotNull Identifier firstPhase, @NotNull Identifier secondPhase,
+			@NotNull ItemConvertible... items) {
+		for (var itemC : items) {
+			ArmorRenderingRegistryImpl.addModelProviderPhaseOrdering(itemC.asItem(), firstPhase, secondPhase);
+		}
+	}
+
+	public static void registerModelProvider(@NotNull ModelProvider provider, @NotNull ItemConvertible... items) {
+		registerModelProvider(DEFAULT_PHASE, provider, items);
+	}
+
+	@Environment(EnvType.CLIENT)
+	public interface TextureProvider {
+		/**
+		 * @param useSecondTexture {@code false} to use helmet/chestplate texture, {@code true} to use leggings/boots texture
+		 * @param suffix           suffix to append to the end of the texture name
+		 */
+		@NotNull Identifier getArmorTexture(@NotNull Identifier texture,
+				@NotNull LivingEntity entity, @NotNull ItemStack stack, @NotNull EquipmentSlot slot,
+				boolean useSecondTexture, @Nullable String suffix);
+	}
+
+	@Environment(EnvType.CLIENT)
+	public interface ModelProvider {
+		@NotNull BipedEntityModel<LivingEntity> getArmorModel(@NotNull BipedEntityModel<LivingEntity> model,
+				@NotNull LivingEntity entity, @NotNull ItemStack stack, @NotNull EquipmentSlot slot);
+	}
```

Comment:
`@NotNull` is implicit. only mark `@Nullable`, unless this changed recently

Reply:
It changed recently

for API we're getting more aggressive with annotations.
And some Kotlin users made good points about those annotations!